### PR TITLE
refactor: migrate private endpoing region to new Atlas SDK

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

_Jira ticket:_ [INTMDB-1101](https://jira.mongodb.org/browse/INTMDB-1101)

Please include a summary of the fix/feature/change, including any relevant motivation and context.

Link to any related issue(s): 

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)
- [ ] This change requires a documentation update

## Manual QA performed:

- [x] cfn invoke for each of CRUDL/cfn test
- [x] Updated resource in  [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples)
- [x] Published to AWS private registry
- [x] Used the template in [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples) to create and update a stack in AWS
- [x] Deleted stack to ensure resources are deleted
- [x] Created multiple resources in same stack
- [x] Validated in Atlas UI
- [x] Included screenshots

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code
- [x] For CFN Resources: I have released by changes in the private registry and proved by change
  works in Atlas

### CFN test
```bash
cfn test --function-name TestEntrypoint --verbose

===== 7 passed, 1 skipped, 16 deselected in 46.19s ====
```

### CFN stack

Screenshot 2023-10-05 at 17 59 22
Screenshot 2023-10-05 at 18 01 35

### CODE REVIEW

File: cfn-resources/private-endpoint-regional-mode/cmd/resource/resource.go [hunk 1]
```diff
@@ -49,16 +47,14 @@ func Create(req handler.Request, prevModel *Model, currentModel *Model) (handler
 		return *errEvent, nil
```
Comment (lantoli) [inline cfn-resources/private-endpoint-regional-mode/cmd/resource/resource.go:47]: remove this log?

File: cfn-resources/private-endpoint-regional-mode/cmd/resource/resource.go [hunk 2]
```diff
@@ -149,25 +137,25 @@ func List(req handler.Request, prevModel *Model, currentModel *Model) (handler.P
...
 		OperationStatus: handler.Success,
 		Message:         "Create Complete",
-		ResourceModel:   regionalPrivateEndpointToModel(*currentModel, regionalizedPrivateEndpointSetting),
+		ResourceModel:   regionalPrivateEndpointToModel(*currentModel),
 	}, nil
 }
 
-func isRegModeSettingExists(currentModel *Model, client *mongodbatlas.Client) bool {
+func isRegModeSettingExists(currentModel *Model, client *util.MongoDBClient) bool {
 	var isExists bool
```
Comment (lantoli) [inline cfn-resources/private-endpoint-regional-mode/cmd/resource/resource.go:157]: I see the logic with isExists was already there, but I think not having that variable will make the function implementation cleaner and easier to understand

**Review by lantoli — DISMISSED** [review]

File: cfn-resources/private-endpoint-regional-mode/cmd/resource/resource.go [hunk 3]
```diff
@@ -177,7 +165,7 @@ func isRegModeSettingExists(currentModel *Model, client *mongodbatlas.Client) bo
 	return isExists
 }
 
-func regionalPrivateEndpointToModel(currentModel Model, regPrivateMode *mongodbatlas.RegionalizedPrivateEndpointSetting) *Model {
+func regionalPrivateEndpointToModel(currentModel Model) *Model {
```
Comment (AgustinBettati) [inline cfn-resources/private-endpoint-regional-mode/cmd/resource/resource.go:168]:
```suggestion
func responseModel(currentModel Model) *Model {
```
was confused with current name, what do you think?

**Review by AgustinBettati — DISMISSED** [review]
LGTM

File: cfn-resources/private-endpoint-regional-mode/cmd/resource/resource.go (hunk 3 above)
Comment (andreaangiolillo) [reply cfn-resources/private-endpoint-regional-mode/cmd/resource/resource.go]: updated

File: cfn-resources/private-endpoint-regional-mode/cmd/resource/resource.go (hunk 2 above)
Comment (andreaangiolillo) [reply cfn-resources/private-endpoint-regional-mode/cmd/resource/resource.go]: updated

File: cfn-resources/private-endpoint-regional-mode/cmd/resource/resource.go (hunk 1 above)
Comment (andreaangiolillo) [reply cfn-resources/private-endpoint-regional-mode/cmd/resource/resource.go:45]: updated

**Review by AgustinBettati — APPROVED** [review]

**Pull request merged**